flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -720,6 +721,27 @@ def clone(node: Node) -> Node:
   return static.merge(state)
 
 
+def iter_nodes(node: tp.Any) -> tp.Iterator[tuple[Path, tp.Any]]:
+  visited: set[int] = set()
+  path_parts: PathParts = ()
+  yield from _iter_nodes(node, visited, path_parts)
+
+
+def _iter_nodes(
+  node: tp.Any, visited: set[int], path_parts: PathParts
```

Comment:
can the `node` arg have a more specific type annotation than `tp.Any`?

Reply:
problem is that a node could be of any type that is registered.